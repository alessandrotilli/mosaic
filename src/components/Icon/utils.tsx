import React from "react";
import MUIAccountCircleIcon from "@material-ui/icons/AccountCircle";
import MUIAddIcon from "@material-ui/icons/Add";
import MUIAppsIcon from "@material-ui/icons/Apps";
import MUIArrowUpwardIcon from "@material-ui/icons/ArrowUpward";
import MUIBlockIcon from "@material-ui/icons/Block";
import MUIBrightness4Icon from "@material-ui/icons/Brightness4";
import MUIBusinessIcon from "@material-ui/icons/Business";
import MUICheckIcon from "@material-ui/icons/Check";
import MUICheckBoxIcon from "@material-ui/icons/CheckBox";
import MUICheckBoxOutlineBlankIcon from "@material-ui/icons/CheckBoxOutlineBlank";
import MUIChevronLeftIcon from "@material-ui/icons/ChevronLeft";
import MUIChevronRightIcon from "@material-ui/icons/ChevronRight";
import MUICloseIcon from "@material-ui/icons/Close";
import MUICropIcon from "@material-ui/icons/Crop";
import MUIDeleteIcon from "@material-ui/icons/Delete";
import MUIDetailsIcon from "@material-ui/icons/Details";
import MUIDraftsIcon from "@material-ui/icons/Drafts";
import MUIEditIcon from "@material-ui/icons/Edit";
import MUIErrorIcon from "@material-ui/icons/Error";
import MUIFilterListIcon from "@material-ui/icons/FilterList";
import MUIFirstPageIcon from "@material-ui/icons/FirstPage";
import MUIGroupWorkIcon from "@material-ui/icons/GroupWork";
import MUIHomeIcon from "@material-ui/icons/Home";
import MUIKeyboardArrowDownIcon from "@material-ui/icons/KeyboardArrowDown";
import MUIKeyboardArrowUpIcon from "@material-ui/icons/KeyboardArrowUp";
import MUILanguageIcon from "@material-ui/icons/Language";
import MUILastPageIcon from "@material-ui/icons/LastPage";
import MUIListIcon from "@material-ui/icons/List";
import MUIMailIcon from "@material-ui/icons/Mail";
import MUIMenuIcon from "@material-ui/icons/Menu";
import MUINotificationsIcon from "@material-ui/icons/Notifications";
import MUIOpenInNew from "@material-ui/icons/OpenInNew";
import MUIPauseIcon from "@material-ui/icons/Pause";
import MUIPlayArrowIcon from "@material-ui/icons/PlayArrow";
import MUIPlaylistAddCheckIcon from "@material-ui/icons/PlaylistAddCheck";
import MUIPowerSettingsNewIcon from "@material-ui/icons/PowerSettingsNew";
import MUIPrintIcon from "@material-ui/icons/Print";
import MUIRefreshIcon from "@material-ui/icons/Refresh";
import MUISaveIcon from "@material-ui/icons/Save";
import MUIScheduleIcon from "@material-ui/icons/Schedule";
import MUISearchIcon from "@material-ui/icons/Search";
import MUISendIcon from "@material-ui/icons/Send";
import MUISettingsIcon from "@material-ui/icons/Settings";
import MUISignalWifi1BarOutlinedIcon from "@material-ui/icons/SignalWifi1BarOutlined";
import MUITimelineIcon from "@material-ui/icons/Timeline";
import MUIVisibilityIcon from "@material-ui/icons/Visibility";
import MUIVisibilityOffIcon from "@material-ui/icons/VisibilityOff";
import MUIWarningIcon from "@material-ui/icons/Warning";
import MUIWifiOffOutlinedIcon from "@material-ui/icons/WifiOffOutlined";
import MUIWifiOutlinedIcon from "@material-ui/icons/WifiOutlined";

import { Icons, IIconsCatalog } from "../../types/Icon";

export const iconsCatalog: IIconsCatalog = {
  [Icons.account]: <MUIAccountCircleIcon />,
  [Icons.add]: <MUIAddIcon />,
  [Icons.apps]: <MUIAppsIcon />,
  [Icons.block]: <MUIBlockIcon />,
  [Icons.brightness]: <MUIBrightness4Icon />,
  [Icons.business]: <MUIBusinessIcon />,
  [Icons.chart_timeline]: <MUITimelineIcon />,
  [Icons.check]: <MUICheckIcon />,
  [Icons.checkbox]: <MUICheckBoxIcon />,
  [Icons.checkbox_empty]: <MUICheckBoxOutlineBlankIcon />,
  [Icons.clock]: <MUIScheduleIcon />,
  [Icons.close]: <MUICloseIcon />,
  [Icons.connection_slow]: <MUISignalWifi1BarOutlinedIcon />,
  [Icons.crop]: <MUICropIcon />,
  [Icons.delete]: <MUIDeleteIcon />,
  [Icons.details]: <MUIDetailsIcon />,
  [Icons.down]: <MUIKeyboardArrowDownIcon />,
  [Icons.edit]: <MUIEditIcon />,
  [Icons.error]: <MUIErrorIcon />,
  [Icons.filter]: <MUIFilterListIcon />,
  [Icons.first]: <MUIFirstPageIcon />,
  [Icons.group]: <MUIGroupWorkIcon />,
  [Icons.home]: <MUIHomeIcon />,
  [Icons.language]: <MUILanguageIcon />,
  [Icons.last]: <MUILastPageIcon />,
  [Icons.left]: <MUIChevronLeftIcon />,
  [Icons.list_check]: <MUIPlaylistAddCheckIcon />,
  [Icons.list]: <MUIListIcon />,
  [Icons.mail_open]: <MUIDraftsIcon />,
  [Icons.mail]: <MUIMailIcon />,
  [Icons.menu]: <MUIMenuIcon />,
  [Icons.notifications]: <MUINotificationsIcon />,
  [Icons.open_new]: <MUIOpenInNew />,
  [Icons.pause]: <MUIPauseIcon />,
  [Icons.play]: <MUIPlayArrowIcon />,
  [Icons.power]: <MUIPowerSettingsNewIcon />,
  [Icons.print]: <MUIPrintIcon />,
  [Icons.refresh]: <MUIRefreshIcon />,
  [Icons.right]: <MUIChevronRightIcon />,
  [Icons.save]: <MUISaveIcon />,
  [Icons.search]: <MUISearchIcon />,
  [Icons.send]: <MUISendIcon />,
  [Icons.settings]: <MUISettingsIcon />,
  [Icons.up_arrow]: <MUIArrowUpwardIcon />,
  [Icons.up]: <MUIKeyboardArrowUpIcon />,
  [Icons.visibility]: <MUIVisibilityIcon />,
  [Icons.visibility_off]: <MUIVisibilityOffIcon />,
  [Icons.warn]: <MUIWarningIcon />,
  [Icons.wifi_off]: <MUIWifiOffOutlinedIcon />,
  [Icons.wifi_on]: <MUIWifiOutlinedIcon />,
};
