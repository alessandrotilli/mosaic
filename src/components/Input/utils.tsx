import React from "react";
import { InputAdornment as MUIInputAdornment } from "@material-ui/core";

import { IInputAdornment } from "../../types/Input";
import IconButton from "../IconButton";

export const getAdornment = (adornment?: IInputAdornment) => {
  if (!adornment) {
    return;
  }

  const { icon, onClick } = adornment;
  return (
    <MUIInputAdornment position="end">
      <IconButton disabled={!onClick} icon={icon} onClick={onClick!} size="small" />
    </MUIInputAdornment>
  );
};
