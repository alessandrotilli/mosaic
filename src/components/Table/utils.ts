export const COLUMN_CHECKBOX_PATH = "checkbox-selection";
export const COLUMN_ROW_ACTIONS_PATH = "row-actions";

export const TOOLBAR_HEIGHT = 64;
export const TOOLBAR_HEIGHT_MOBILE = 56;

export const PAGINATION_TOOLBAR_HEIGHT = 64;
