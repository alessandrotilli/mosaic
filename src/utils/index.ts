import { ChangeEvent, MouseEvent } from "react";

export interface ISubpart {
  label: string;
  value?: Function;
}

export interface ISubpartSuffix {
  label: string;
  suffix: string;
}

export type ISubpartMap<T extends string = string> = {
  [key in T]: ISubpart;
};

export const DATA_CY_SUFFIX_SEPARATOR = "-";

/**
 * Gets composed dataCy for components with subparts
 * @param dataCy the dataCy prefix
 * @param subpart the subpart to consider
 * @param args optional arguments for subpart evaluation
 */
export const getComposedDataCy = (dataCy: string, subpart: ISubpart, args?: any): string => {
  const { label, value } = subpart;
  return `${dataCy}${DATA_CY_SUFFIX_SEPARATOR}${!value ? slugify(label) : value(args)}`;
};

/**
 * Gets all composed dataCy for all listed subparts
 * @param subpartMap the map of all subparts to consider
 */
export const getAllComposedDataCy = (subpartMap: ISubpartMap): ISubpartSuffix[] =>
  Object.values(subpartMap).map(({ label, value }) => ({
    label,
    suffix: !value ? slugify(label) : value(),
  }));

/**
 * Gets an object property by path, if path is nested is executed recursively
 * @param object the source object
 * @param path the desired path
 */
export const getObjectProperty = (object: any, path: string): any => {
  if (!object) {
    return undefined;
  }

  if (!path.includes(".")) {
    return object[path];
  }

  const pathParts = path.split(".");
  const basePath = pathParts[0];
  if (!object[basePath]) {
    return undefined;
  }

  const baseObject = object[basePath];
  return getObjectProperty(baseObject, path.replace(`${basePath}.`, ""));
};

/**
 * Creates slug or kebab-case string from value
 * @param value the value to slugify
 */
export const slugify = (value: string) =>
  value
    .toLowerCase()
    .replace(/\s+/g, "-")
    .replace(/[^\w-]+/g, "");

/**
 * Suppresses default behaviour and propagation of a given event
 * @param event the event to suppress
 */
export const suppressEvent = (event: ChangeEvent<any> | MouseEvent<any, MouseEvent> | null) => {
  if (!event) {
    return;
  }

  event.preventDefault();
  event.stopPropagation();
};
