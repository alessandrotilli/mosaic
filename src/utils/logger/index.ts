/* eslint-disable no-console */
export type ILogLevel = "error" | "info" | "warn";

const MOSAIC_TAG = "[@melfore/mosaic]";

/**
 * Logs message for given level and component
 * @param level the level of the message to log (e.g. "error")
 * @param component the component subject of the log
 * @param message the message of the log
 */
const logger = (level: ILogLevel, component: string, message: string) => {
  const text = `${MOSAIC_TAG} ${component} - ${message}`;

  switch (level) {
    case "error":
      console.error(text);
      return;
    case "info":
      console.info(text);
      return;
    case "warn":
      console.warn(text);
      return;
  }
};

/**
 * Logs message for error level and component
 * @param component the component subject of the log
 * @param message the message of the log
 */
export const logError = (component: string, message: string) => logger("error", component, message);

/**
 * Logs message for info level and component
 * @param component the component subject of the log
 * @param message the message of the log
 */
export const logInfo = (component: string, message: string) => logger("info", component, message);

/**
 * Logs message for warn level and component
 * @param component the component subject of the log
 * @param message the message of the log
 */
export const logWarn = (component: string, message: string) => logger("warn", component, message);
