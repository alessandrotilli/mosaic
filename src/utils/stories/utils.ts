export const DOCS_BODY_CLASS = "mosaic-documentation-body";
export const DOCS_CODELINE_CLASS = "mosaic-documentation-code-line";
export const DOCS_CODEBLOCK_CLASS = "mosaic-documentation-code-block";
export const DOCS_TITLE_CLASS = "mosaic-documentation-title";

export const DOCS_PAGE_STYLE = `
  .${DOCS_BODY_CLASS},
  .${DOCS_TITLE_CLASS} {
    color: #333333 !important;
    font-family: Arial !important;
    cursor: text;
  }
  .${DOCS_BODY_CLASS} {
    font-size: 14px;
    line-height: 24px;
  }
  .${DOCS_TITLE_CLASS} {
    margin: 20px 0 8px;
    padding: 0;
    position: relative;
  }
  h2.${DOCS_TITLE_CLASS} {
    border-bottom: 1px solid rgba(0,0,0,.1);
    font-size: 24px;
    padding-bottom: 4px;
  }
  h3.${DOCS_TITLE_CLASS} {
    font-size: 20px;
  }
  .${DOCS_BODY_CLASS} code.${DOCS_CODELINE_CLASS},
  .${DOCS_BODY_CLASS} code.${DOCS_CODEBLOCK_CLASS} {
    background-color: #F8F8F8;
    font-family: "Operator Mono","Fira Code Retina","Fira Code","FiraCode-Retina","Andale Mono","Lucida Console",Consolas,Monaco,monospace;
    font-size: 13px;
    padding: 4px;
  }
  .${DOCS_BODY_CLASS} code.${DOCS_CODEBLOCK_CLASS} {
    background-color: #F8F8F8;
    color: rgba(51,51,51,0.9);
    display: flex;
    padding: 8px;
    white-space: pre-line;
  }
  .icon-wrapper {
    align-items: center;
    display: flex;
  }
  .icon-wrapper > span {
    color: #333333 !important;
    font-family: Arial !important;
    font-size: 12px;
    margin-left: 4px;
  }
  .stories-wrapper {
    display: flex;
    flex-wrap: wrap;
  }
  .stories-wrapper > * {
    margin: 8px;
  }
  .typography-wrapper {
    display: flex;
    flex-direction: column;
    width: 100%;
  }
`;
