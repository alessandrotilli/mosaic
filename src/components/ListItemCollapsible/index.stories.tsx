import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";

import { getAllComposedDataCy } from "../../utils";
import getDocsPage from "../../utils/stories";

import ListItemCollapsible, { DATA_CY_DEFAULT, SUBPARTS_MAP } from ".";

export default {
  title: "Display/ListItemCollapsible",
  component: ListItemCollapsible,
  parameters: {
    docs: {
      ...getDocsPage({
        basedOn: {
          label: "MUI Collapse Component",
          url: "https://v4.mui.com/components/transitions/#collapse",
        },
        component: "ListItemCollapsible",
        e2eTestInfo: {
          dataCyDefault: DATA_CY_DEFAULT,
          subpartsSuffixes: getAllComposedDataCy(SUBPARTS_MAP),
        },
      }),
    },
  },
} as ComponentMeta<typeof ListItemCollapsible>;

const Template: ComponentStory<typeof ListItemCollapsible> = (args) => (
  <ListItemCollapsible {...args} dataCy={DATA_CY_DEFAULT} />
);

export const Primary = Template.bind({});
Primary.args = {
  content: "Collapsible Content",
  header: "Header",
};

export const Dense = Template.bind({});
Dense.args = {
  ...Primary.args,
  dense: true,
};

export const Loading = Template.bind({});
Loading.args = {
  ...Primary.args,
  loading: true,
};

export const Open = Template.bind({});
Open.args = {
  ...Primary.args,
  open: true,
};

export const Selected = Template.bind({});
Selected.args = {
  ...Primary.args,
  selected: true,
};

// export const Basic = () => <ListItemCollapsible header={<Typography>Collapsible List Item</Typography>} />;

// export const CustomStyle = () => (
//   <ListItemCollapsible
//     header={<Typography>Custom Style</Typography>}
//     style={{ backgroundColor: "red", color: "white" }}
//   />
// );

// export const Dense = () => <ListItemCollapsible dense header={<Typography>Collapsible List Item</Typography>} />;

// export const Loading = () => <ListItemCollapsible loading header={<Typography>Collapsible List Item</Typography>} />;

// export const Selected = () => <ListItemCollapsible selected header={<Typography>Collapsible List Item</Typography>} />;

// export const Open = () => (
//   <ListItemCollapsible open selected header={<Typography>Collapsible List Item</Typography>}>
//     <div style={{ border: "1px solid #ccc", padding: "16px" }}>
//       <Typography children="Collapsible content is now expanded" />
//     </div>
//   </ListItemCollapsible>
// );
