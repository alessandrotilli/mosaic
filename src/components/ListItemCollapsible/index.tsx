import React, { FC, memo, PropsWithChildren, useMemo } from "react";
import { Collapse as MUICollapse } from "@mui/material";

import { Icons } from "../../types/Icon";
import { IListItemCollapsible } from "../../types/ListItem";
import { getComposedDataCy } from "../../utils";
import ListItem from "../ListItem";

export const DATA_CY_DEFAULT = "list-item-collapsible";

export const SUBPARTS_MAP = {
  header: {
    label: "Header",
  },
  collapse: {
    label: "Collapse",
  },
};

const ListItemCollapsible: FC<PropsWithChildren<IListItemCollapsible>> = ({
  children,
  content,
  dataCy = DATA_CY_DEFAULT,
  dense = false,
  header,
  loading = false,
  onClick,
  open = false,
  openTimeout = "auto",
  selected = false,
  style,
  unmountContent = false,
}) => {
  const icon = useMemo(() => (open ? Icons.up : Icons.down), [open]);

  const collapsibleContent = useMemo(() => content || children, [content, children]);

  return (
    <>
      <ListItem
        dataCy={getComposedDataCy(dataCy, SUBPARTS_MAP.header)}
        dense={dense}
        icon={icon}
        loading={loading}
        onClick={onClick}
        selected={selected}
        style={style}
      >
        {header}
      </ListItem>
      <MUICollapse
        data-cy={getComposedDataCy(dataCy, SUBPARTS_MAP.collapse)}
        in={open}
        timeout={openTimeout}
        unmountOnExit={unmountContent}
      >
        {collapsibleContent}
      </MUICollapse>
    </>
  );
};

export default memo(ListItemCollapsible);
