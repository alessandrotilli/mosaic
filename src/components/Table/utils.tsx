import React, { forwardRef, Fragment } from "react";
import { Action as MTActionType, Column as MTColumnType, Options as MTOptionsType } from "material-table";
import { Icons, IconSize } from "../../types/Icon";
import { TableActionPosition, ITableAction, ITableColumn } from "../../types/Table";
import Button from "../Button";
import Icon from "../Icon";
import IconButton from "../IconButton";
import { getDataCyForSubComponent, suppressEvent } from "../../utils";

// For default props refer to Options section here https://material-table.com/#/docs/all-props
export const DEFAULT_TABLE_OPTIONS: MTOptionsType = {
  debounceInterval: 500,
  draggable: false,
  emptyRowsWhenPaging: false,
};

export const actionAdapter = (action: ITableAction): MTActionType<object> => {
  const {
    callback,
    label,
    disabled = false,
    hidden = false,
    icon = undefined,
    position = TableActionPosition.default,
  } = action;
  return {
    disabled,
    hidden,
    icon: `${label}$${icon}`,
    onClick: (event, data) => {
      if (event) {
        suppressEvent(event);
      }
      callback(data);
    },
    position,
  };
};

export const actionComponentAdapter = (props: any, dataCy: string) => {
  const { data, action } = props;
  const { disabled, hidden, icon, position, onClick }: MTActionType<object> = action;
  if (hidden) {
    return null;
  }

  const displaysButton = position === TableActionPosition.default;
  const labelAndIconName = icon as string;
  const label = labelAndIconName.split("$")[0];
  const iconName = labelAndIconName.split("$")[1] as Icons;
  return (
    <Fragment>
      {displaysButton && (
        <Fragment>
          <span>&nbsp;&nbsp;&nbsp;&nbsp;</span>
          <Button
            dataCy={getDataCyForSubComponent(dataCy, `action-${label}`)}
            disabled={disabled}
            icon={!iconName ? undefined : { name: iconName }}
            label={label}
            onClick={() => onClick(undefined, data)}
          />
        </Fragment>
      )}
      {!displaysButton && (
        <Fragment>
          <IconButton
            dataCy={getDataCyForSubComponent(dataCy, `action-${label}`)}
            disabled={disabled}
            icon={iconName}
            onClick={() => onClick(undefined, data)}
            size={IconSize.small}
          />
          <span>&nbsp;&nbsp;</span>
        </Fragment>
      )}
    </Fragment>
  );
};

export const columnAdapter = (column: ITableColumn): MTColumnType<object> => {
  const { label, path, render, width } = column;
  return {
    ...(!!width && ({ width } as object)),
    field: path,
    render,
    title: label,
  };
};

export const iconAdapter = (name: Icons, size = IconSize.default) =>
  forwardRef((props: any, ref: any) => {
    return <Icon forwarded={{ ...props, ref }} name={name} size={size} />;
  });
