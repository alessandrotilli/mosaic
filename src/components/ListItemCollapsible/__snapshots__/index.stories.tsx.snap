// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display/ListItemCollapsible Dense smoke-test 1`] = `
<li class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-6lnq02-MuiListItem-root"
    data-cy="list-item-collapsible-header"
    style="cursor: pointer;"
>
  <div class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root">
    <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1wnp694-MuiSvgIcon-root"
         focusable="false"
         aria-hidden="true"
         viewbox="0 0 24 24"
         data-testid="KeyboardArrowDownIcon"
         data-cy="list-item-collapsible-header-icon"
    >
      <path d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z">
      </path>
    </svg>
  </div>
  <div class="MuiListItemText-root MuiListItemText-dense css-tlelie-MuiListItemText-root"
       data-cy="list-item-collapsible-header-content"
  >
    Header
  </div>
</li>
<div class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
     data-cy="list-item-collapsible-collapse"
     style="min-height: 0px;"
>
  <div class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper">
    <div class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner">
      Collapsible Content
    </div>
  </div>
</div>
`;

exports[`Display/ListItemCollapsible Loading smoke-test 1`] = `
<li class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    data-cy="list-item-collapsible-header"
    style="cursor: default;"
>
  <div class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root">
    <span class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-153n0d3-MuiSkeleton-root"
          style="width: 24px; height: 24px;"
    >
    </span>
  </div>
  <div class="MuiListItemText-root css-tlelie-MuiListItemText-root"
       data-cy="list-item-collapsible-header-content-loading"
  >
    <span class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root">
    </span>
  </div>
</li>
<div class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
     data-cy="list-item-collapsible-collapse"
     style="min-height: 0px;"
>
  <div class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper">
    <div class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner">
      Collapsible Content
    </div>
  </div>
</div>
`;

exports[`Display/ListItemCollapsible Open smoke-test 1`] = `
<li class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    data-cy="list-item-collapsible-header"
    style="cursor: pointer;"
>
  <div class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root">
    <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1wnp694-MuiSvgIcon-root"
         focusable="false"
         aria-hidden="true"
         viewbox="0 0 24 24"
         data-testid="KeyboardArrowUpIcon"
         data-cy="list-item-collapsible-header-icon"
    >
      <path d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6z">
      </path>
    </svg>
  </div>
  <div class="MuiListItemText-root css-tlelie-MuiListItemText-root"
       data-cy="list-item-collapsible-header-content"
  >
    Header
  </div>
</li>
<div class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
     data-cy="list-item-collapsible-collapse"
     style="min-height: 0px;"
>
  <div class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper">
    <div class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner">
      Collapsible Content
    </div>
  </div>
</div>
`;

exports[`Display/ListItemCollapsible Primary smoke-test 1`] = `
<li class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    data-cy="list-item-collapsible-header"
    style="cursor: pointer;"
>
  <div class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root">
    <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1wnp694-MuiSvgIcon-root"
         focusable="false"
         aria-hidden="true"
         viewbox="0 0 24 24"
         data-testid="KeyboardArrowDownIcon"
         data-cy="list-item-collapsible-header-icon"
    >
      <path d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z">
      </path>
    </svg>
  </div>
  <div class="MuiListItemText-root css-tlelie-MuiListItemText-root"
       data-cy="list-item-collapsible-header-content"
  >
    Header
  </div>
</li>
<div class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
     data-cy="list-item-collapsible-collapse"
     style="min-height: 0px;"
>
  <div class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper">
    <div class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner">
      Collapsible Content
    </div>
  </div>
</div>
`;

exports[`Display/ListItemCollapsible Selected smoke-test 1`] = `
<li class="MuiListItem-root MuiListItem-gutters MuiListItem-padding Mui-selected css-1p823my-MuiListItem-root"
    data-cy="list-item-collapsible-header"
    style="cursor: pointer;"
>
  <div class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root">
    <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1wnp694-MuiSvgIcon-root"
         focusable="false"
         aria-hidden="true"
         viewbox="0 0 24 24"
         data-testid="KeyboardArrowDownIcon"
         data-cy="list-item-collapsible-header-icon"
    >
      <path d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z">
      </path>
    </svg>
  </div>
  <div class="MuiListItemText-root css-tlelie-MuiListItemText-root"
       data-cy="list-item-collapsible-header-content"
  >
    Header
  </div>
</li>
<div class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
     data-cy="list-item-collapsible-collapse"
     style="min-height: 0px;"
>
  <div class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper">
    <div class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner">
      Collapsible Content
    </div>
  </div>
</div>
`;
